grails-app/views/speciesListItem/_facet.gsp

```diff
@@ -59,12 +59,12 @@
                                         <tr>
                                             <g:if test="${isProperty}">
                                                 <td><g:link action="list" id="${params.id}"
-                                                            params="${[fq: sl.buildFqList(fqs: fqs, fq: "kvp ${arr[0]}:${arr[1]}"), max: params.max]}">${arr[2] ?: arr[1]}</g:link></td>
+                                                            params="${[fq: sl.buildFqList(fqs: fqs, fq: "kvp ${arr[0]}:${arr[1]}"), q: (params.q) ?: null, max: params.max]}">${arr[2] ?: arr[1]}</g:link></td>
```

Comment:
I'm not sure that `q: (params.q) ?: null` would be any different to using just `q: params.q` (both give `null`) unless you absolutely want an empty string to be changed to null (if an empty string can come in via servlet).

Reply:
updated